resources/models.py

```diff
@@ -615,33 +616,57 @@ def get_queryset(self):
             - If there are leases associated to the berth
             - If any lease ends during the current or the last season (previous year)
                 + If a lease ends during the current season:
-                    * It needs to have a "valid" status (DRAFTED, OFFERED, PAID)
+                    * It needs to have a "valid" status (DRAFTED, OFFERED, PAID, ERROR)
                 + If a lease ended during the last season:
+                    * It should not have been renewed for the next season
                     * It needs to have a "valid" status (PAID)
-                    * It needs to have renew automatically set
         """
         from leases.models import BerthLease
 
-        current_season_end = calculate_berth_lease_end_date()
+        season_end = calculate_season_end_date()
+        current_date = today().date()
+
+        # If today is before the season ends but during the same year
+        if current_date < season_end and current_date.year == season_end.year:
+            last_year = current_date.year - 1
+        else:
+            last_year = current_date.year
+
+        in_current_season = Q(end_date__gte=season_end)
+        in_last_season = Q(end_date__year=last_year)
```

Comment:
How does the code handle the case where:
* Lease is in the current season
* Lease.status is TERMINATED
* lease.end_date is equal to season_end

Is the berth considered available in that case? It should not, am I correct? Is that case in the scope of this PR at all?

Reply:
* When the lease is in the current season ([resources/tests/test_resources_models.py#L123-L132](https://github.com/City-of-Helsinki/berth-reservations/blob/ven-1137/mark-refused-berths-available/resources/tests/test_resources_models.py#L123-L132)), the berth is **not** available
* When the lease is terminated ([resources/tests/test_resources_models.py#L135-L144](https://github.com/City-of-Helsinki/berth-reservations/blob/ven-1137/mark-refused-berths-available/resources/tests/test_resources_models.py#L135-L144)), the berth is available
* When the lease ends with the season ([resources/models.py#L635](https://github.com/City-of-Helsinki/berth-reservations/blob/ven-1137/mark-refused-berths-available/resources/models.py#L635)), since it checks `gte` it should include the ones which end with the season, so the berth is available

Reply:
OK!